pkg/security/apiserver/admission/sccadmission/admission.go

```diff
@@ -151,33 +193,49 @@ loop:
 			break loop
 		case apiequality.Semantic.DeepEqual(pod, podCopy):
 			// if we don't allow mutation, only use the validated pod if it didn't require any spec changes
 			allowedPod = podCopy
 			allowingProvider = provider
-			glog.V(6).Infof("pod %s (generate: %s) validated against provider %s without mutation", pod.Name, pod.GenerateName, provider.GetSCCName())
+			glog.V(4).Infof("pod %s (generate: %s) validated against provider %s without mutation", pod.Name, pod.GenerateName, provider.GetSCCName())
 			break loop
 		default:
-			glog.V(6).Infof("pod %s (generate: %s) validated against provider %s, but required mutation, skipping", pod.Name, pod.GenerateName, provider.GetSCCName())
+			glog.V(4).Infof("pod %s (generate: %s) validated against provider %s, but required mutation, skipping", pod.Name, pod.GenerateName, provider.GetSCCName())
 		}
 	}
 
-	if allowedPod != nil {
-		*pod = *allowedPod
-		// annotate and accept the pod
-		glog.V(4).Infof("pod %s (generate: %s) validated against provider %s", pod.Name, pod.GenerateName, allowingProvider.GetSCCName())
-		if pod.ObjectMeta.Annotations == nil {
-			pod.ObjectMeta.Annotations = map[string]string{}
-		}
-		pod.ObjectMeta.Annotations[allocator.ValidatedSCCAnnotation] = allowingProvider.GetSCCName()
-		return nil
+	if allowedPod == nil || allowingProvider == nil {
+		return nil, "", validationErrs, nil
 	}
+	return allowedPod, allowingProvider.GetSCCName(), validationErrs, nil
+}
 
-	// we didn't validate against any security context constraint provider, reject the pod and give the errors for each attempt
-	glog.V(4).Infof("unable to validate pod %s (generate: %s) against any security context constraint: %v", pod.Name, pod.GenerateName, validationErrs)
-	return admission.NewForbidden(a, fmt.Errorf("unable to validate against any security context constraint: %v", validationErrs))
+func shouldIgnore(a admission.Attributes) (bool, error) {
+	if a.GetResource().GroupResource() != kapi.Resource("pods") {
+		return true, nil
+	}
+	if len(a.GetSubresource()) != 0 {
+		return true, nil
+	}
+
+	_, ok := a.GetObject().(*kapi.Pod)
+	// if we can't convert then fail closed since we've already checked that this is supposed to be a pod object.
+	// this shouldn't normally happen during admission but could happen if an integrator passes a versioned
+	// pod object rather than an internal object.
+	if !ok {
+		return false, admission.NewForbidden(a, fmt.Errorf("object was marked as kind pod but was unable to be converted: %v", a.GetObject()))
+	}
+
+	// if this is an update, see if we are only updating the ownerRef.  Garbage collection does this
+	// and we should allow it in general, since you had the power to update and the power to delete.
+	// The worst that happens is that you delete something, but you aren't controlling the privileged object itself
+	if a.GetOldObject() != nil && rbacregistry.IsOnlyMutatingGCFields(a.GetObject(), a.GetOldObject(), kapihelper.Semantic) {
```

Comment:
should we also be checking that the operation is `UPDATE`?

Reply:
> should we also be checking that the operation is UPDATE?

This is a straight move.  I didn't check.  We have concerns about existing behavior?

Reply:
tracked it down to a 1.12-era PR fixing admission for create-on-update cases (https://github.com/kubernetes/kubernetes/pull/65572#discussion_r199762096). this  is fine for 1.11, but we need to sweep before 1.12 to make sure we're not using `GetOldObject() != nil` when we really mean `GetOperation() == UPDATE`